lib/Runtime/Types/TypePath.cpp

```diff
@@ -8,18 +8,28 @@ namespace Js {
 
     TypePath* TypePath::New(Recycler* recycler, uint size)
     {
+        Assert(size <= MaxPathTypeHandlerLength);
         size = max(size, InitialTypePathSize);
-        size = PowerOf2Policy::GetSize(size);
+        
 
         if (PHASE_OFF1(Js::TypePathDynamicSizePhase))
         {
             size = MaxPathTypeHandlerLength;
         }
+        else
+        {
+            size = PowerOf2Policy::GetSize(size - TYPE_PATH_ALLOC_GRANULARITY_GAP);
+            if (size < MaxPathTypeHandlerLength)
+            {
+                size += TYPE_PATH_ALLOC_GRANULARITY_GAP;
+            }
+        }
 
         Assert(size <= MaxPathTypeHandlerLength);
 
         TypePath * newTypePath = RecyclerNewPlusZ(recycler, sizeof(PropertyRecord *) * size, TypePath);
-        newTypePath->pathSize = (uint16)size;
+        // Allocate enough space for the "next" for the TinyDictionary;
+        newTypePath->data = RecyclerNewPlusLeafZ(recycler, size, TypePath::Data, (uint8)size);
```

Comment:
Ah sorry mistakenly read the change (thought the removed code is newly added code)